blockchain/system/kip113.go

```diff
@@ -0,0 +1,147 @@
+// Copyright 2023 The klaytn Authors
+// This file is part of the klaytn library.
+//
+// The klaytn library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The klaytn library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the klaytn library. If not, see <http://www.gnu.org/licenses/>.
+
+package system
+
+import (
+	"math/big"
+	"sort"
+	"strings"
+
+	"github.com/klaytn/klaytn/accounts/abi/bind"
+	"github.com/klaytn/klaytn/common"
+	contracts "github.com/klaytn/klaytn/contracts/system_contracts"
+	"github.com/klaytn/klaytn/crypto/bls"
+)
+
+type BlsPublicKeyInfo struct {
+	PublicKey []byte
+	Pop       []byte
+}
+
+type (
+	BlsPublicKeyInfos map[common.Address]BlsPublicKeyInfo
+	AllocKip113Init   = BlsPublicKeyInfos
+)
+
+func AllocKip113(init AllocKip113Init) map[common.Hash]common.Hash {
```

Comment:
For the dynamic type (`bytes` and `string`), refactoring may be considered so that the generic calcuator address all of inputs  such as

```go
func calcDynTyp(baseSlot interface{}, b []byte) (common.Hash, map[common.Hash]common.Hash) {
    var (
        WORD_SIZE = 32
        ret       = make(map[common.Hash]common.Hash)
        l         = len(b)
        dynLen    = lpad32(WORD_SIZE)
    )
    if l >= WORD_SIZE {
        dynLen = lpad32(l*2 + 1)
        done := 0

        for i := WORD_SIZE; i <= l; i += WORD_SIZE {
            key := calcStructSlot(baseSlot, i/WORD_SIZE-1)
            val := common.BytesToHash(b[i/WORD_SIZE-1 : i])
            ret[key] = val
            done += WORD_SIZE
        }
        if l-done > 0 {
            padded := append(b[done:], make([]byte, l-done)...)
            key := calcStructSlot(baseSlot, len(ret))
            val := common.BytesToHash(padded)
            ret[key] = val
        }
    } else {
        var val common.Hash
        copy(val[:], b)

        val[WORD_SIZE-1] = byte(l * 2)
        ret[baseSlot.(common.Hash)] = val
    }
    return dynLen, ret
}
```

With this, the inner loop can be shortened with consistent format
```go
dynLen, ret := calcDynTyp(pupKeySlot, info.PublicKey)
storage[pupKeySlot] = dynLen
for k, v := range ret {
    storage[k] = v
}

dynLen, ret = calcDynTyp(popKeySlot, info.Pop)
storage[popKeySlot] = dynLen
for k, v := range ret {
    storage[k] = v
}
```

Reply:
@hyunsooda Nice. I will try.

Reply:
@hyunsooda Implemented 6830206 PTAL.